storm-client/src/jvm/org/apache/storm/daemon/worker/WorkerState.java

```diff
@@ -717,11 +719,19 @@ private Assignment getLocalAssignment(IStormClusterState stormClusterState, Stri
 
         IWaitStrategy backPressureWaitStrategy = IWaitStrategy.createBackPressureWaitStrategy(topologyConf);
         Map<List<Long>, JCQueue> receiveQueueMap = new HashMap<>();
+
         for (List<Long> executor : executors) {
             List<Integer> taskIds = StormCommon.executorIdToTasks(executor);
+            int taskId = taskIds.get(0);
+            String compId;
+            if (taskId == -1) {
```

Comment:
I would  use `SYSTEM_TASK_ID` instead of `-1`

Reply:
Yes, updated